### PR TITLE
[NTDLL_APITEST][KMTEST] Add tests for FileEndOfFileInformation

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Add tests for FileEndOfFileInformation

## Tests
✅ Test WHS: https://reactos.org/testman/compare.php?ids=94854,94862
✅ Test Windows 2003 x64: https://reactos.org/testman/compare.php?ids=94850,94863
✅ Test KVM x86: https://reactos.org/testman/compare.php?ids=95077,95079,95086,95093
✅ Test KVM x64: https://reactos.org/testman/compare.php?ids=95096,95104

### CODE REVIEW

File: modules/rostests/kmtests/ntos_io/IoFilesystem.c [hunk 1]
```diff
@@ -221,6 +221,15 @@ TestAllInformation(VOID)
     if (FileAllInfo)
         KmtFreeGuarded(FileAllInfo);
 
+    PFILE_END_OF_FILE_INFORMATION FileEofInfo;
+    Length = sizeof(FILE_END_OF_FILE_INFORMATION);
+    Status = QueryFileInfo(FileHandle, (PVOID*)&FileEofInfo, &Length, FileEndOfFileInformation);
+    // Checked build: STATUS_INVALID_INFO_CLASS, Free build: STATUS_INVALID_PARAMETER
+    ok(Status == STATUS_INVALID_PARAMETER || Status == STATUS_INVALID_INFO_CLASS, "Wrong Status = %lx\n", Status);
```
Comment (binarymaster) [inline modules/rostests/kmtests/ntos_io/IoFilesystem.c:228]: Heh, it's interesting that the returned status depends on chk/fre.

**Review by binarymaster — APPROVED** [review]

File: modules/rostests/apitests/ntdll/NtQueryInformationFile.c [hunk 2]
```diff
@@ -20,4 +20,36 @@ START_TEST(NtQueryInformationFile)
...
+
+    /* Query FileEndOfFileInformation */
+    FILE_END_OF_FILE_INFORMATION EndOfFileInformation;
+    EndOfFileInformation.EndOfFile.QuadPart = 0xdeaddead;
+    Status = NtQueryInformationFile(hFile,
+                                    NULL,
+                                    &EndOfFileInformation,
+                                    sizeof(EndOfFileInformation),
+                                    FileEndOfFileInformation);
+    ok_hex(Status, STATUS_INVALID_INFO_CLASS);
```
Comment (oleg-dubinskiy) [inline modules/rostests/apitests/ntdll/NtQueryInformationFile.c:51]: Shouldn't this function return `STATUS_INVALID_PARAMETER` for free build type in user mode too? Like you did in the kmtest part.

**Review by DarkFire01 — APPROVED** [review]

File: modules/rostests/apitests/ntdll/NtQueryInformationFile.c (hunk 2 above)
Comment (tkreuzer) [reply modules/rostests/apitests/ntdll/NtQueryInformationFile.c:51]: The user mode check is the same, only in kernel mode checked builds have an extra check that returns early with a different error code.

Comment (oleg-dubinskiy) [reply modules/rostests/apitests/ntdll/NtQueryInformationFile.c:51]: Fine for me then. Thanks.

**Review by oleg-dubinskiy — APPROVED** [review]

File: modules/rostests/apitests/ntdll/NtQueryInformationFile.c [hunk 3]
```diff
@@ -20,4 +20,36 @@ START_TEST(NtQueryInformationFile)
...
+                               NULL,
+                               OPEN_EXISTING,
+                               FILE_ATTRIBUTE_NORMAL,
+                               NULL);
+    ok(hFile != INVALID_HANDLE_VALUE, "CreateFileA failed\n");
+    if (hFile == INVALID_HANDLE_VALUE)
+        return;
+
+    /* Query FileEndOfFileInformation */
+    FILE_END_OF_FILE_INFORMATION EndOfFileInformation;
```
Comment (HBelusca) [inline modules/rostests/apitests/ntdll/NtQueryInformationFile.c:44]: I'm not a big fan of variables declarations in the middle of code in C.

File: modules/rostests/apitests/ntdll/NtQueryInformationFile.c [hunk 4]
```diff
@@ -20,4 +20,36 @@ START_TEST(NtQueryInformationFile)
     Status = NtQueryInformationFile(NULL, NULL, NULL, 0, 0x80000000);
     ok(Status == STATUS_INVALID_INFO_CLASS ||
        ntv6(Status == STATUS_NOT_IMPLEMENTED), "Status = %lx\n", Status);
+
+    /* Get the full path of the current executable */
+    CHAR Path[MAX_PATH];
+    DWORD Length = GetModuleFileNameA(NULL, Path, MAX_PATH);
```
Comment (HBelusca) [inline modules/rostests/apitests/ntdll/NtQueryInformationFile.c:26]:
```suggestion
    DWORD Length = GetModuleFileNameA(NULL, Path, _countof(Path));
```

File: modules/rostests/kmtests/ntos_io/IoFilesystem.c [hunk 5]
```diff
@@ -221,6 +221,15 @@ TestAllInformation(VOID)
     if (FileAllInfo)
         KmtFreeGuarded(FileAllInfo);
 
+    PFILE_END_OF_FILE_INFORMATION FileEofInfo;
+    Length = sizeof(FILE_END_OF_FILE_INFORMATION);
```
Comment (HBelusca) [inline modules/rostests/kmtests/ntos_io/IoFilesystem.c:225]:
```suggestion
    Length = sizeof(*FileEofInfo);
```

File: modules/rostests/apitests/ntdll/NtQueryInformationFile.c (hunk 3 above)
Comment (tkreuzer) [reply modules/rostests/apitests/ntdll/NtQueryInformationFile.c:44]: And I am not a big fan of style nit-picking in PRs.

**Pull request merged**